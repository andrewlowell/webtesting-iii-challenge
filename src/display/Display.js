import React from 'react';

const Display = ({ closed, locked }) => {
  const closedClass = `led ${closed ? 'red-led' : 'green-led'}`;
  const lockedClass = `led ${locked ? 'red-led' : 'green-led'}`;

  return (
    <div className="display panel">
      <div className={lockedClass} data-testid="disp-lock">{locked ? 'Locked' : 'Unlocked'}</div>
      <div className={closedClass} data-testid="disp-open">{closed ? 'Closed' : 'Open'}</div>
    </div>
  );
};

Display.defaultProps = {
  closed: false,
  locked: false,
};

export default Display;
