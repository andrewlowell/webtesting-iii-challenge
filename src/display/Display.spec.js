import React from "react";
import { render, fireEvent, cleanup, getByTestId } from "react-testing-library";
import renderer from "react-test-renderer";
import "jest-dom/extend-expect";

import Display from "./Display";
import Dashboard from "../dashboard/Dashboard";

afterEach(cleanup);

describe("Display tests!", () => {
  it("renders!", () => {
    render(<Display />);
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Display />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should display the correct colors with the correct words on initial load", () => {
    const { getByTestId } = render(<Display />);

    expect(getByTestId('disp-lock')).toHaveClass("green-led");
    expect(getByTestId('disp-open')).toHaveClass("green-led");
  });

  it("should display the correct colors in response to event triggers", () => {
    const { getByTestId, getByText } = render(<Dashboard />);
    expect(getByTestId('disp-open')).toHaveClass("green-led");
    expect(getByTestId('disp-lock')).toHaveClass("green-led");

    const closeButton = getByText(/Close Gate/);
    fireEvent.click(closeButton);

    expect(getByTestId('disp-open')).toHaveClass("red-led");

    const lockButton = getByText(/Lock Gate/);
    fireEvent.click(lockButton);

    expect(getByTestId('disp-lock')).toHaveClass("red-led");

    const unlockButton = getByText(/Unlock Gate/);
    fireEvent.click(unlockButton);

    expect(getByTestId('disp-lock')).toHaveClass("green-led");

    const openButton = getByText(/Open Gate/);
    fireEvent.click(openButton);

    expect(getByTestId('disp-open')).toHaveClass("green-led");
  });
});