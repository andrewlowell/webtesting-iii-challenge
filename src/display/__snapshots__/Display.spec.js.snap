// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display tests! matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="disp-lock"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="disp-open"
  >
    Open
  </div>
</div>
`;
